smart-contracts/assembly/contracts/dns/dns.ts

```diff
@@ -256,3 +267,81 @@ export function addWebsiteToBlackList(binaryArgs: StaticArray<u8>): void {
   Storage.set(blackListKey, new Args().add(newList).serialize());
   generateEvent(`Domain name ${websiteName} added to blackList`);
 }
+
+/**
+ * Retrieves the array of blacklisted keys.
+ *
+ * @returns The array of blacklisted keys.
+ */
+export function getBlacklisted(): string[] {
+  // Create a key for the blacklist in storage
+  const blackListKey = new Args().add('blackList').serialize();
+
+  // Deserialize the blacklisted keys from storage, if it exists
+  const blacklistedKeys = Storage.has(blackListKey)
+    ? new Args(Storage.get(blackListKey)).nextNativeTypeArray<string>().unwrap()
+    : [];
+
+  return blacklistedKeys;
+}
+
+/**
+ * Appends multiple website names to the blacklist.
+ *
+ * @param binaryArgs - Website names in a binary format using Args.
+ */
+export function addWebsitesToBlackList(binaryArgs: StaticArray<u8>): void {
+  // Ensure that the caller is the contract owner
+  onlyOwner();
+
+  // Create a key for the blacklist in storage
+  const blackListKey = new Args().add('blackList').serialize();
+
+  // Extract the website names from binaryArgs and unwrap them into an array
+  const websiteNames = new Args(binaryArgs)
+    .nextNativeTypeArray<string>()
+    .unwrap();
+
+  // Retrieve the current blacklisted keys
+  const existingBlacklist = getBlacklisted();
+
+  // Create a Set to ensure uniqueness of website names
+  const blacklistSet = new Set<string>();
+
+  // Merge the existing blacklist with the new website names
+  const mergedWebsiteNames = websiteNames.concat(existingBlacklist);
+
+  // Add each website name to the blacklist Set
+  // Iterators are not Implemented workaround done here
+  // to avoid compilation errors AS100
```

Comment:
i was not expecting this kind of comment.
I was thinking to add a comment here on why we use "set" instead of keeping string[]
=> because Set garanti that every element is unique. so here we handle the case where a name is already blacklisted withou adding it twice